api/api.go

```diff
@@ -751,11 +752,14 @@ func (api *Server) GetEvmTransfersByActionHash(ctx context.Context, in *iotexapi
 
 	actHash, err := hash.HexStringToHash256(in.ActionHash)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	transfers, err := api.systemLogIndexer.GetEvmTransfersByActionHash(actHash)
 	if err != nil {
+		if strings.Contains(err.Error(), "key = ") && strings.Contains(err.Error(), "doesn't exist") {
```

Comment:
This check is too hacky. Can we use errors.Cause() to check the root cause? If you don't wrap the error in systemLogIndexer.GetEvmTransfersByActionHash(actHash), please wrap it with a ErrNotExist so you can check here.